distributed/tests/test_publish.py

```diff
@@ -37,6 +37,24 @@ def test_publish_simple(s, a, b):
     yield f.close()
 
 
+@gen_cluster(client=False)
+def test_publish_non_string_key(s, a, b):
+    c = yield Client((s.ip, s.port), asynchronous=True)
+    f = yield Client((s.ip, s.port), asynchronous=True)
+
+    for name in [('a', 'b'), 9.0, 8]:
+        data = yield c.scatter(range(3))
+        out = yield c.publish_dataset(data, name=name)
+        assert name in s.extensions['publish'].datasets
+        assert isinstance(s.extensions['publish'].datasets[name]['data'], Serialized)
+
+        datasets = yield c.scheduler.publish_list()
+        assert name in datasets
+
+    yield c.close()
+    yield f.close()
```

Comment:
This should probably use a try-finally block to clean up the clients even if the test fails in the future.

Reply:
Should this be done by the @gen_cluster rather than relying on the test to do it?

Reply:
I'd say thats out of scope for this pr.  I can make another one for that, since that might be handy to simplify some of the test suite